File: python/exercise_202.py
class Solution:
    def isHappy(self, n: int) -> bool:
        sum = 0
        a = list(str(n))
        print(a)
        results = []
        while sum != 1:
            sum = 0
            for num in a:
                sum += int(num)*int(num)
            print(sum)
            a = list(str(sum))
            if sum in results:
                return False
            results.append(sum)
        return True

File: java/hashsetOrHashmap/exercise_202.java
class Solution {
    public boolean isHappy(int n) {
        Set<Integer> seen = new HashSet<>();
        while(!seen.contains(n)){
            seen.add(n);
            int aux = 0;
            while(n!=0){
                aux+=(int)(n%10)*(int)(n%10);
                n/=10;
            }
            n = aux;
            if(n==1){
                return true;
            }
        }
        return false;
    }
}